packages/core/saved-objects/core-saved-objects-api-server-internal/src/lib/apis/types.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { Logger } from '@kbn/logging';
+import type {
+  ISavedObjectTypeRegistry,
+  SavedObjectsExtensions,
+  ISavedObjectsSerializer,
+} from '@kbn/core-saved-objects-server';
+import type { IKibanaMigrator, IndexMapping } from '@kbn/core-saved-objects-base-server-internal';
+import type { RepositoryHelpers } from './helpers';
+import type { RepositoryEsClient } from '../repository_es_client';
+
+export interface ApiExecutionContext {
+  registry: ISavedObjectTypeRegistry;
+  helpers: RepositoryHelpers;
+  extensions: SavedObjectsExtensions;
+  client: RepositoryEsClient;
+  allowedTypes: string[];
+  serializer: ISavedObjectsSerializer;
+  migrator: IKibanaMigrator;
+  logger: Logger;
+  mappings: IndexMapping;
+}
```

Comment:
The 'execution context' (terrible name, if someone wants to suggest something better please do) that is passed down to all API execution handlers.

I initially though about defining the strict subset of properties used by each handler, but in the end, it was way more verbose, and forced to touch more code when adding new properties to the context, so I think this generic / single type approach is better.